tests/test_truncation.py

```diff
@@ -55,3 +59,95 @@ def test_truncate():
         # so the degeneracy criteria shoulb be ignored and we should just keep 6 states
         mask, norm_new, TE = truncation.truncate(S_shuffled, pars.copy())
         is_expected_S(S[:6], S_shuffled[mask])
+
+
+@pytest.mark.parametrize('use_eig_based_svd', [
+    (False),
+    (True),
+])
+@pytest.mark.filterwarnings("ignore:_eig_based_svd is nonsensical on CPU!!")
+def test_decompose_theta_qr_based(use_eig_based_svd):
+    # Evolve state to obtain evolved tensors
+    model_params = {
+        'J': 1. , 'g': 1.,
+        'L': 16,
+        'bc_MPS': 'finite',    
+        }
+    M = TFIChain(model_params)
+    psi = MPS.from_lat_product_state(M.lat, [['up']])
+    options = {
+        'dt': 0.1,
+        'N_steps': 1,
+        'order': 1,
+        'approximation': 'I',
+        'cbe_min_block_increase': 1,
+        'cbe_expand': float(0.1),
+        'use_eig_based_svd': use_eig_based_svd,
+        'compression_method': 'variationalQR',
+        'trunc_params': {
+            'chi_max': 50,
+            'svd_min': 10**(-12),
+        }
+    }
+    eng = mpo_evolution.ExpMPOEvolution(psi, M, options)
+    for i in range(15):
+        eng.run()
+    
+    # Fabricate dummy theta (not actually updated)
+    S = psi.get_SL(8)
+    old_T_L = psi.get_B(8, 'B').ireplace_label('p','p0')
+    old_T_R = psi.get_B(9, 'B').ireplace_label('p','p1')
+    theta = npc.tensordot(old_T_L.scale_axis(S, axis='vL'), old_T_R, ['vL','vR'])
+    theta = theta.combine_legs([['vL','p0'],['p1','vR']])
```

Comment:
For both cases (`move_right=False` and `move_right=True`) the tensors 'old_T_L` and `old_T_R` are in `'B'` form. It does not matter if `S` does not change the charge. However as mentioned here, this can be a problem.